app/views/permissions/show.html.erb

```diff
@@ -11,7 +11,7 @@
         </tr>
       </thead>
       <tbody>
-        <% if @permission.fetch('group_permissions', []).blank? %>
+        <% if @permission.fetch('group_permissions', []).blank? || @permission.fetch('group_permissions', []).all? { |group| group.fetch('permissions', []).blank? } %>
```

Comment:
Unfortunately, I think you need to handle legacy data cases where the permissions might have `create`, `update`, or `delete` permissions, and still have it display 'No groups listed'.